linera-views/src/common.rs

```diff
@@ -190,57 +256,228 @@ pub trait KeyValueStoreClient {
+                    if let Some(ref mut value) = value {
+                        for _ in 0..*count {
+                            let segment: Vec<u8> = segments.get(pos).unwrap().clone().unwrap();
+                            value.extend(segment);
+                            pos += 1;
+                        }
+                    }
+                }
+            }
+        }
+        Ok(big_values)
+    }
 
-    /// Finds the `key` matching the prefix. The prefix is not included in the returned keys.
-    async fn find_keys_by_prefix(&self, key_prefix: &[u8]) -> Result<Self::Keys, Self::Error>;
+    async fn find_keys_by_prefix(&self, key_prefix: &[u8]) -> Result<Self::Keys, Self::Error> {
+        let mut keys = Vec::new();
+        for big_key in self
+            .client
+            .find_keys_by_prefix(key_prefix)
+            .await?
+            .iterator()
+        {
+            let big_key = big_key?;
+            let len = big_key.len();
+            let last_byte = big_key.get(len - 1).unwrap();
+            if last_byte == &0 {
+                let key = big_key[0..len - 1].to_vec();
+                keys.push(key);
+            }
+        }
+        Ok(keys)
+    }
 
-    /// Finds the `(key,value)` pairs matching the prefix. The prefix is not included in the returned keys.
     async fn find_key_values_by_prefix(
         &self,
         key_prefix: &[u8],
-    ) -> Result<Self::KeyValues, Self::Error>;
-
-    /// Writes the `batch` in the database with `base_key` the base key of the entries for the journal.
-    async fn write_batch(&self, batch: Batch, base_key: &[u8]) -> Result<(), Self::Error>;
-
-    /// Clears any journal entry that may remain.
-    /// The journal is located at the `base_key`.
-    async fn clear_journal(&self, base_key: &[u8]) -> Result<(), Self::Error>;
-
-    /// Reads a single `key` and deserialize the result if present.
-    async fn read_key<V: DeserializeOwned>(&self, key: &[u8]) -> Result<Option<V>, Self::Error>
-    where
-        Self::Error: From<bcs::Error>,
-    {
-        from_bytes_opt(self.read_key_bytes(key).await?)
+    ) -> Result<Self::KeyValues, Self::Error> {
+        let mut key_values = Vec::new();
+        let mut key = Vec::new();
+        let mut big_value = Vec::new();
+        let mut count = 0;
+        let mut pos = 0;
+        for result in self
+            .client
+            .find_key_values_by_prefix(key_prefix)
+            .await?
+            .iterator()
+        {
+            let (big_key, value) = result?;
+            let len = big_key.len();
+            let last_byte = big_key.get(len - 1).unwrap();
+            if last_byte == &0 {
+                let first_value = value.first();
+                if first_value.is_none() {
+                    return Err(DatabaseConsistencyError::ZeroLengthValue.into());
+                }
+                let first_value = *first_value.unwrap();
```

Comment:
```suggestion
                let Some(&first_byte) = value.first() else {
                    return Err(DatabaseConsistencyError::ZeroLengthValue.into());
                }
```

Reply:
Ok, this passes. Needs to add a final `;` though.